### ISSUE TITLE
channelHandlers has null object

### ISSUE THREAD

**Issue by KamToHung** [apache/dubbo#11357]
- [x] I have searched the [issues](https://github.com/apache/dubbo/issues) of this repository and believe that this is not a duplicate.

### Environment

* Dubbo version: 3.2
* Operating System version: macOS Big Sur
* Java version: 8

### Steps to reproduce this issue

1. simple test
```java
public class NPETest {

    protected WorldHandler handler = new WorldHandler();

    @Test
    void handlerIsNull() throws RemotingException {
        Exchangers.bind("",null, handler);
    }

}
```
2. handler is not null in this step
```java
    public ChannelHandlerDispatcher(ChannelHandler... handlers) {
        this(handlers == null ? null : Arrays.asList(handlers));
    }
```
3. channelHandlers will add a null handler
```java
    public ChannelHandlerDispatcher(Collection<ChannelHandler> handlers) {
        if (CollectionUtils.isNotEmpty(handlers)) {
            this.channelHandlers.addAll(handlers);
        }
    }
```
4. connected, disconnected, sent and caught method will NPE

### Expected Behavior

channelHandlers have no null object

### Actual Behavior

channelHandlers have null object

### reason
```java
   public ChannelHandlerDispatcher(ChannelHandler... handlers) {
        this(handlers == null ? null : Arrays.asList(handlers));
    }

   public ChannelHandlerDispatcher(Collection<ChannelHandler> handlers) {
        if (CollectionUtils.isNotEmpty(handlers)) {
            this.channelHandlers.addAll(handlers);
        }
    }
```
if handlers is null, it will cast to ChannelHandler and use the first method then handlers is not null

we can filter the null object when add to channelHandlers

Can I fix it?

**Comment by AlbumenJ** [comment]
Please go ahead

**Comment by CrazyHZM** [comment]
It has been solved in #11360.
